rbac/management/relation_replicator/sync_replicator.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""RelationReplicator which writes to the outbox table."""
+
+import json
+import logging
+
+
+import grpc
+from django.conf import settings
+from google.rpc import error_details_pb2
+from grpc_status import rpc_status
+from kessel.relations.v1beta1 import relation_tuples_pb2
+from kessel.relations.v1beta1 import relation_tuples_pb2_grpc
+from management.relation_replicator.relation_replicator import RelationReplicator, ReplicationEvent
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class SyncReplicator(RelationReplicator):
+    """Replicates relations via the Relations API over gRPC."""
+
+    def replicate(self, event: ReplicationEvent):
+        """Replicate the given event to Kessel Relations via the gRPC API."""
+        self._write_relationships(event.add)
```

Comment:
There is support only for add relations, is it intentional ?

Reply:
Yeah, this is just to port what the old migrator code did, which is only add relations (which makes sense for the migrator). We can add more as needed later.